File: algorithm/python/BOJ1259.py
# problem : https://www.acmicpc.net/problem/1259

import sys

answer = list()
input_number = sys.stdin.readline().rstrip()
while input_number != "0":
    reverse_number = input_number[::-1]

    if reverse_number == input_number:
        answer.append("yes")
    else:
        answer.append("no")

    input_number = sys.stdin.readline().rstrip()

for item in answer:
    print(item)

File: algorithm/typescript/BOJ1259.ts
// https://www.acmicpc.net/problem/1259

import fs = require("fs");

const input: string[] = fs.readFileSync("/dev/stdin").toString().trim().split("\n");
const answer: string[] = new Array();

input.forEach((data) => {
  if (data === "0") {
    return;
  }

  const reversedString: string = data.split("").reverse().join("");

  if (reversedString === data) {
    answer.push("yes");
  } else {
    answer.push("no");
  }
});

console.log(answer.join("\n"));
